examples/python/ml/flax_llama7b_split/README.md

```diff
@@ -0,0 +1,131 @@
+
+    ```sh
+    cd path_to_EasyLM/EasyLM/models/llama
+    python convert_hf_to_easylm.py  \
+       --checkpoint_dir     path_to_llama_weights    \
+       --output_file path_to_outputfile  \
+       --model_size 7b \
+       --streaming
+    ```
+
+    Move the python file to EasyLM if you do not use the fork fork created for transformer split.
+
+    ```sh
+    cp path-to-llama_model_split_transformer_py path_to_EasyLM/EasyLM/models/llama
+    ```
+
+3. Launch SPU backend runtime
+
+    ```sh
+    bazel run -c opt //examples/python/utils:nodectl -- --config `pwd`/examples/python/ml/flax_llama7b_split/3pc.json up
+    ```
+
+    or
+    （recommended）
+
+    ```sh
+    cd examples/python/utils
+    python nodectl.py  --config `pwd`/examples/python/ml/flax_llama7b_split/3pc.json up
+    ```
+
+4. Run `flax_llama7b_split` example
+
+    ```sh
+    bazel run -c opt //examples/python/ml/flax_llama7b_split -- --config `pwd`/examples/python/ml/flax_llama7b_split/3pc.json
+    ```
+
+    or（recommended）
+
+    ```sh
+    cd examples/python/ml/flax_llama7b_split
+    python flax_llama7b_split.py --config `pwd`/examples/python/ml/flax_llama7b_split/3pc.json
+    ```
+
+    and you can get the following results from our example:
+
+    ```md
+    ------
+    Run on CPU
+    Q: What is the largest animal?
+    A: The largest animal is the blue whale.
+    generate on CPU: 256.08824276924133 seconds
+
+    ------
+    Run on SPU
+    [0000-00-00 00:00:00.000] [info] [thread_pool.cc:30] Create a fixed thread pool with size 127
+    Q: What is the largest animal?
+    A: The largest animal is the blue whale.
+    generate  on SPU: 812.9427680969238 seconds
+    ```
+    RAM peak: 64.5888GB
+
+    And If you set token_num to 30, you can get the following results:
+    ```sh
+   ------
+    Run on CPU
+    Q: What is the largest animal?
+    A: The largest animal is the blue whale.
+    Q: What is the smallest animal?
+    A: The smallest animal is the bacterium.
+    generate on CPU: 837.0810837745667 seconds
+
+    ------
+    Run on SPU
+    [0000-00-00 00:00:00.000] [info] [thread_pool.cc:30] Create a fixed thread pool with size 127
+    Q: What is the largest animal?
+    A: The largest animal is the blue whale.
+    Q: What is the smallest animal?
+    A: The smallest animal is the bacterium.
+    generate  on SPU: 2760.5035905838013 seconds
+    ```
+4. Privacy Security Warning
+
+In this example, our main motivation is to reduce the hardware and time resource costs of [Llama-7B](https://research.facebook.com/publications/llama-open-and-efficient-foundation-language-models/) model inference using the SPU. Therefore, spu is only used for inference on the middle blocks of the model.  Its privacy protection capability for the original data is weaker when using spu for inference on the entire Llama-7B model. It may be vulnerable to Model Inversion Attacks known in Split Learning as follows:
```

Comment:
Supplement the split strategy? Like 
- Client: Embedding layer + 1-2 Block
- Mid: 3rd Block
- ..